src/omotes_simulator_core/adapter/transforms/transform_utils.py

```diff
@@ -15,6 +15,65 @@
 
 """File containing utility functions for the transforms."""
 from typing import Dict, List
+from enum import Enum
+from dataclasses import dataclass
+
+
+class PortType(Enum):
+    """Enum to define the type of port."""
+
+    IN = 1
+    OUT = 2
+
+
+@dataclass
+class Port:
+    """Dataclass to hold the port information."""
+
+    port_id: str
+    port_name: str
+    port_type: PortType
+
+
+def sort_ports(connected_ports: list[Port]) -> list[str]:
+    """Sort the ports of the asset based on the port type.
+
+    The sort order is Inport, Outport, Starting with the first inport, then the first outport
+    and so on.
+
+    :param connected_ports: List of tuples with the port id, name and the port type.
+    :return: List of port ids sorted by port type.
+    """
+    in_ports = [port for port in connected_ports if port.port_type == PortType.IN]
+    out_ports = [port for port in connected_ports if port.port_type == PortType.OUT]
+    if len(in_ports) != len(out_ports):
+        raise ValueError("The number of in ports and out ports are not equal")
+    sorted_port_list = []
+    for in_port, out_port in zip(in_ports, out_ports):
+        sorted_port_list.append(in_port)
+        sorted_port_list.append(out_port)
+    if len(sorted_port_list) == 4:
```

Comment:
Is this always the case? We should check with ESDL because you can manually add ports. 

Maybe check for asset type?

Reply:
I could add a error when a asset has another number of ports then 2 or 4, only these values are supported

Reply:
Yeah that would be best

Reply:
created issue #204 to add this.